sdk/core/src/services/BloomVaultStorageService.ts

```diff
@@ -0,0 +1,293 @@
+    this._platformEncryptionTools = platformEncryptionTools
+    this._vaultApiService = new BloomVaultApiService(options)
+  }
+
+  /* istanbul ignore next: private function */
+  private _getVaultKeys() {
+    const { seed } = this._keysService.decryptSeed()
+
+    const privateKey = bip32.fromSeed(seed).derivePath(jolocomIdentityKey).privateKey
+    const privateKeyHex = privateKey.toString('hex')
+
+    const publicKey = privateToPublic(privateKey)
+    const addressHex = publicToAddress(publicKey).toString('hex')
+
+    return { addressHex, privateKeyHex }
+  }
+
+  /* istanbul ignore next: ethereumjs-util */
+  private _signByVaultKeys(message: string, privateKey: string) {
+    const sig = ecsign(hashPersonalMessage(Buffer.from(message)), Buffer.from(privateKey, 'hex'))
+
+    return toRpcSig(sig.v, sig.r, sig.s)
+  }
+
+  /* istanbul ignore next: private function */
+  private async _authorizeVcVault(storageRegion: string) {
+    const { addressHex, privateKeyHex } = this._getVaultKeys()
+    const didEth = `did:ethr:0x${addressHex}`
+
+    const {
+      body: { token },
+    } = await this._vaultApiService.requestAuthToken({
+      storageRegion,
+      did: didEth,
+    })
+
+    const signature = this._signByVaultKeys(token, privateKeyHex)
+
+    await this._vaultApiService.validateAuthToken({
+      storageRegion,
+      accessToken: token,
+      signature,
+      did: didEth,
+    })
+
+    return token
+  }
+
+  /* istanbul ignore next: private function */
+  private async _deleteCredentialByIndex(accessToken: string, index: number, storageRegion: string): Promise<void> {
+    try {
+      // NOTE: deletes the data objects associated with the included access token
+      //       and the included IDs starting with :start and ending with :end inclusive
+      //       https://github.com/hellobloom/bloom-vault#delete-datastartend
+      await this._vaultApiService.deleteCredentials({ accessToken, storageRegion, start: index, end: index })
+    } catch (error) {
+      throw new SdkErrorFromCode('COR-0', {}, error)
+    }
+  }
+
+  /* istanbul ignore next: private function */
+  private async _fetchEncryptedCredentialsWithPagination(
+    paginationOptions: PaginationOptions,
+    accessToken: string,
+    storageRegion: string,
+  ) {
+    try {
+      const { body: blobs } = await this._vaultApiService.getCredentials({
+        accessToken,
+        storageRegion: storageRegion,
+        start: paginationOptions.skip,
+        end: paginationOptions.skip + paginationOptions.limit - 1,
+      })
+
+      if (blobs.length === 0) {
+        return undefined
+      }
+
+      return blobs.filter((blob) => blob.cyphertext !== null)
+    } catch (error) {
+      if (error.httpStatusCode === 404) {
+        throw new SdkErrorFromCode('COR-14', {}, error)
+      } else {
+        throw error
+      }
+    }
+  }
+
+  /* istanbul ignore next: private function */
+  private async _fetchEncryptedCredentials(
```

Comment:
This method does not seem to be used anywhere?

I wonder why IDE did not highlight it as unused (since it's private)

Reply:
it was highlighted, I just missed it, thank you